src/main/java/edu/ie3/util/io/FileIOUtils.java

```diff
@@ -141,6 +157,495 @@ public static BufferedWriter getBufferedWriter(
+
+    /* Get the archive file size */
+    long archiveSize = archive.toFile().length();
+
+    /* Create the target folder */
+    try {
+      Files.createDirectories(targetDirectory);
+    } catch (IOException e) {
+      throw new FileException("Cannot create target directory '" + targetDirectory + "'.", e);
+    }
+
+    /* Monitor amount of entries and their size for safety reasons */
+    int entries = 0;
+    long size = 0;
+    try (InputStream fileInputStream = Files.newInputStream(archive);
+        BufferedInputStream bufferedInputStream = new BufferedInputStream(fileInputStream);
+        GzipCompressorInputStream gzipInputStream =
+            new GzipCompressorInputStream(bufferedInputStream);
+        TarArchiveInputStream tarInputStream = new TarArchiveInputStream(gzipInputStream)) {
+      ArchiveEntry archiveEntry;
+      while ((archiveEntry = tarInputStream.getNextEntry()) != null) {
+        /* Control the total amount of entries */
+        entries++;
+        if (entries > MAX_AMOUNT_OF_ENTRIES)
+          throw new IOException(
+              "The archive contains too many entries and is therefore possibly malicious.");
+
+        /* Control the size of extracted archive files */
+        long uncompressedSize = archiveEntry.getSize();
+        if (uncompressedSize == ArchiveEntry.SIZE_UNKNOWN)
+          throw new IOException(
+              "Unknown uncompressed file size of '" + archiveEntry.getName() + "'");
+        size += uncompressedSize;
+        if (size > MAX_SIZE_UNCOMPRESSED)
+          throw new IOException(
+              "Uncompressed size of archive exceeds permissible "
+                  + (MAX_SIZE_UNCOMPRESSED / 1024 / 1024)
+                  + " MB. Possibly malicious archive");
+
+        /* Control the compression ratio */
+        if (1 - (double) archiveSize / size > MAX_COMPRESSION_RATIO)
+          throw new IOException(
+              "Compression ratio exceeds its maximum permissible value "
+                  + (MAX_COMPRESSION_RATIO * 100)
+                  + " %. Possibly malicious archive");
+
+        handleZipEntrySafely(archiveEntry, targetDirectory, tarInputStream);
+      }
+    } catch (IOException ex) {
+      throw new FileException("Unable to extract from '" + archive + "'.", ex);
+    }
+
+    return targetDirectory;
+  }
+
+  /**
+   * Extracts the given zipped file to a file with the same name that the zipped file has beneath
+   * the target directory.
+   *
+   * @param zippedFile Compressed gzip file to extract
+   * @param target Path to the target folder
+   * @return Path of the file, where the content is extracted to
+   * @throws FileException If the zipped file is not in a well shape or the target path doesn't meet
+   *     the requirements
+   * @see <a href="https://mkyong.com/java/how-to-decompress-file-from-gzip-file/">MyKong
+   *     Tutorial</a>
+   */
+  public static Path extractFile(Path zippedFile, Path target) throws FileException {
```

Comment:
Same here.

Reply:
Sure, I will change the function extractFile(Path, Path) accordingly.